Add mul tests with sparse sample inputs

pearu opened the pull request:
This PR implements sparse sample inputs and error inputs for mul OpInfo.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #98292
* #100394
* __->__ #100393



cc @alexsamardzic @nikitaved @cpuhrsch @amjames @bhosmer

pearu committed:
Add mul tests with sparse sample inputs

[ghstack-poisoned]

pearu committed:
Update on "Add mul tests with sparse sample inputs"




cc alexsamardzic nikitaved cpuhrsch amjames bhosmer

[ghstack-poisoned]

pearu committed:
Update on "Add mul tests with sparse sample inputs"

This PR implements sparse sample inputs and error inputs for mul OpInfo.




cc alexsamardzic nikitaved cpuhrsch amjames bhosmer

[ghstack-poisoned]

amjames reviewed (approved):
One nit, otherwise looks great.

pearu commented on torch/testing/_internal/opinfo/definitions/sparse.py:
```diff
@@ -447,6 +480,289 @@ def error_inputs_sparse_reduction_sum(op_info, device, layout, **kwargs):
+
+def _validate_sample_input_elementwise_binary_sparse_mul(sample):
+    # NOTE: When fixing a failing sample case, remove the
+    #       corresponding if-block
```
I'll do that in a follow-up to unblock landing the fixup in https://github.com/pytorch/pytorch/pull/98292

pearu committed:
Update on "Add mul tests with sparse sample inputs"

This PR implements sparse sample inputs and error inputs for mul OpInfo.




cc alexsamardzic nikitaved cpuhrsch amjames bhosmer

[ghstack-poisoned]

pearu committed:
Update on "Add mul tests with sparse sample inputs"

This PR implements sparse sample inputs and error inputs for mul OpInfo.




cc alexsamardzic nikitaved cpuhrsch amjames bhosmer

[ghstack-poisoned]

pearu committed:
Update on "Add mul tests with sparse sample inputs"

This PR implements sparse sample inputs and error inputs for mul OpInfo.




cc alexsamardzic nikitaved cpuhrsch amjames bhosmer

[ghstack-poisoned]

pearu commented:
@pytorchbot merge
